joda-money/src/main/java/com/fasterxml/jackson/datatype/jodamoney/MoneySerializer.java

```diff
@@ -1,23 +1,25 @@
 package com.fasterxml.jackson.datatype.jodamoney;
 
-import java.io.IOException;
-import java.math.BigDecimal;
-import java.math.RoundingMode;
-
-import org.joda.money.Money;
-
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonToken;
 import com.fasterxml.jackson.core.type.WritableTypeId;
 import com.fasterxml.jackson.databind.SerializerProvider;
 import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import org.joda.money.Money;
+
+import java.io.IOException;
+
+import static java.util.Objects.requireNonNull;
 
 public class MoneySerializer extends JodaMoneySerializerBase<Money>
 {
     private static final long serialVersionUID = 1L;
 
-    public MoneySerializer() {
+    private final AmountConverter<?> amountConverter;
```

Comment:
As with deserializer, for 2.x backwards compatibility would be good to keep deprecated no-args constructor (to be removed from 3.0).

Reply:
ditto